src/Components/DynamicFields.res

```diff
@@ -183,285 +236,95 @@ let make = (
-      | Bank => Some(selectedBank !== "" || bankNames->Belt.Array.length === 0)
-      | PhoneNumber => Some(phone.value !== "")
-      | StateAndCity => Some(state.value !== "" && city.value !== "")
-      | CountryAndPincode(countryArr) =>
-        Some(
-          (country !== "" || countryArr->Belt.Array.length === 0) &&
-            postalCode.isValid->Belt.Option.getWithDefault(false),
-        )
-      | AddressCity => Some(city.value !== "")
-      | AddressPincode => postalCode.isValid
-      | AddressState => Some(state.value !== "")
-      | BlikCode => Some(blikCode.value !== "")
-      | Currency(currencyArr) => Some(currency !== "" || currencyArr->Belt.Array.length === 0)
-      | AddressLine2
-      | SpecialField(_)
-      | InfoElement
-      | None =>
-        Some(true)
-      }->Belt.Option.getWithDefault(false)
-    }, true)
-    setAreRequiredFieldsValid(._ => areRequiredFieldsValid)
-
-    let areRequiredFieldsEmpty = fieldsArr->Js.Array2.reduce((acc, paymentMethodFields) => {
-      acc ||
-      switch paymentMethodFields {
-      | Email => email.value === ""
-      | FullName => fullName.value === ""
-      | Country => country === "" && countryNames->Belt.Array.length > 0
-      | AddressCountry(countryArr) => country === "" && countryArr->Belt.Array.length > 0
-      | BillingName => billingName.value === ""
-      | AddressLine1 => line1.value === ""
-      | Bank => selectedBank === "" && bankNames->Belt.Array.length > 0
-      | StateAndCity => city.value === "" || state.value === ""
-      | CountryAndPincode(countryArr) =>
-        (country === "" && countryArr->Belt.Array.length > 0) || postalCode.value === ""
-      | PhoneNumber => phone.value === ""
-      | AddressCity => city.value === ""
-      | AddressPincode => postalCode.value === ""
-      | AddressState => state.value === ""
-      | BlikCode => blikCode.value === ""
-      | Currency(currencyArr) => currency === "" && currencyArr->Belt.Array.length > 0
-      | AddressLine2
-      | SpecialField(_)
-      | InfoElement
-      | None => false
-      }
-    }, false)
-    setAreRequiredFieldsEmpty(._ => areRequiredFieldsEmpty)
-    None
-  }, (
-    fieldsArr,
-    currency,
-    fullName.value,
-    country,
-    billingName.value,
-    line1.value,
-    (
-      email,
-      line2.value,
-      selectedBank,
-      phone.value,
-      city.value,
-      postalCode.value,
-      state.value,
-      blikCode.value,
-    ),
-  ))
+  DynamicFieldsUtils.requiredFieldsEmptyAndValidHook(
+    ~fieldsArr,
+    ~country,
+    ~countryNames,
+    ~selectedBank,
+    ~bankNames,
+    ~currency,
+    ~email,
+    ~fullName,
+    ~billingName,
+    ~line1,
+    ~line2,
+    ~phone,
+    ~state,
+    ~city,
+    ~postalCode,
+    ~blikCode,
+    ~isCardValid,
+    ~isExpiryValid,
+    ~isCVCValid,
+    ~cardNumber,
+    ~cardExpiry,
+    ~cvcNumber,
+    ~setAreRequiredFieldsValid,
+    ~setAreRequiredFieldsEmpty,
```

Comment:
The above recoil states can be pulled by the hook itself instead of passing it as props